projects/core/src/occ/services/occ-endpoints.service.ts

```diff
@@ -58,52 +88,116 @@ export class OccEndpointsService {
    * @param endpoint Endpoint suffix
    */
   getOccEndpoint(endpoint: string): string {
-    if (!this.config?.backend?.occ) {
-      return '';
-    }
-    endpoint = this.config.backend.occ.endpoints?.[endpoint];
+    endpoint = this.getRawEndpointValue(endpoint);
 
-    if (
-      !endpoint.startsWith('/') &&
-      !this.config.backend.occ.prefix.endsWith('/')
-    ) {
-      endpoint = '/' + endpoint;
-    }
-
-    return (
-      this.config.backend.occ.baseUrl +
-      this.config.backend.occ.prefix +
-      endpoint
-    );
+    return this.getEndpoint(endpoint, { baseUrl: true, prefix: true });
   }
 
   /**
-   * Returns base OCC endpoint (baseUrl + prefix + baseSite)
+   * Returns base OCC endpoint (baseUrl + prefix + baseSite) by if no parameters are specified
+   *
+   * @param endpointComponents The prefixes to add to the url (baseUrl, prefix, baseSite)
    */
-  getBaseEndpoint(): string {
+  getBaseEndpoint(endpointComponents?: EndpointComponents): string {
     if (!this.config?.backend?.occ) {
       return '';
     }
 
-    return (
-      (this.config.backend.occ.baseUrl || '') +
-      this.config.backend.occ.prefix +
-      this.activeBaseSite
-    );
+    if (!endpointComponents) {
+      return (
+        (this.config.backend.occ.baseUrl || '') +
+        this.config.backend.occ.prefix +
+        this.activeBaseSite
+      );
+    } else {
+      return (
```

Comment:
Can we have an utility that takes care about all the `/` between segments?
Something similar to `path.join`?